src/devices/Gpio/Drivers/OrangePiZeroDriver.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Gpio.Drivers
+{
+    /// <summary>
+    /// A GPIO driver for the Orange Pi Zero.
+    /// </summary>
+    /// <remarks>
+    /// SoC: Allwinner H2+ (sun8iw7p1)
+    /// </remarks>
+    public class OrangePiZeroDriver : Sun8iw7p1Driver
+    {
+        /// <summary>
+        /// Orange Pi Zero has 17 GPIO pins.
+        /// </summary>
+        protected override int PinCount => 17;
+
+        /// <summary>
+        /// Converts a board pin number to the driver's logical numbering scheme.
+        /// </summary>
+        /// <param name="pinNumber">The board pin number to convert.</param>
+        /// <returns>The pin number in the driver's logical numbering scheme.</returns>
+        protected override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber)
+        {
+            return pinNumber switch
+            {
+                3 => MapPinNumber('A', 12),
```

Comment:
(optional) you might instead want to create array with 27 elements and map it to number or -1 if not present... switch will check it one by one and be relatively slow.